backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -1054,19 +1046,36 @@ abstract class ObjCExportHeaderGenerator internal constructor(
         return ObjCExportedInterface(generatedClasses, extensions, topLevel, headerLines, namer, mapper)
     }
 
+    fun getExportStubs(): ObjCExportedStubs =
+        ObjCExportedStubs(classForwardDeclarations, protocolForwardDeclarations, stubs)
+
     protected abstract fun reportWarning(text: String)
 
     protected abstract fun reportWarning(method: FunctionDescriptor, text: String)
 
     protected open fun getAdditionalImports(): List<String> = emptyList()
 
-
-    fun translateModule(): List<Stub<*>> {
+    fun translateModule() {
         // TODO: make the translation order stable
         // to stabilize name mangling.
+        translateBaseDeclarations()
+        translatePackageFragments()
+        translateExtraClasses()
```

Comment:
Does this actually add any stubs for `ObjcExportHeaderGeneratorMobile(restrictToLocalModules = true)`?
If no and this is intended, then it makes some of sense to use only `translatePackageFragments` in this case (considering https://github.com/JetBrains/kotlin-native/pull/4498#discussion_r518068321), don't patch `translateExtraClasses` and change the name and the meaning of  `shouldTranslateExtraClass` to something like `shouldTranslateExtraClassEarly`. The implementation and other usages also suggest the latter.

Reply:
Verified: `translateExtraClasses` is still adding stubs for `ObjcExportHeaderGeneratorMobile(restrictToLocalModules = true)`. Seems to happen very rarely, but I am seeing an inner classes being translated in this way.

Reply:
Ok.
But in any case, we didn't supposed it to be this way.

Reply:
Could you please clarify? Would you like me to file a bug about the inner classes not being generated by `translatePackageFragments`?

Reply:
No, I don't think this is necessary for now.
Adding a comment makes more sense.

Reply:
Added doc comment.